packages/cojson-transport-ws/src/index.ts

```diff
@@ -102,24 +120,77 @@ export function createWebSocketPeer({
         }
     });
 
+    const highPriorityQueue: {msg: SyncMessage, promise:Promise<void>, resolve: () => void}[] = [];
+    const lowPriorityQueue: {msg: SyncMessage,  promise:Promise<void>, resolve: () => void}[] = [];
+
+    let processingActive = false;
+
+    async function processQueue() {
+        if (processingActive) {
+            return;
+        }
+
+        processingActive = true;
+
+        if (websocket.readyState !== 1) {
+            await websocketOpen;
+        }
+
+        while (highPriorityQueue.length > 0 || lowPriorityQueue.length > 0) {
+            if (websocket.bufferedAmount > BUFFER_LIMIT) {
+                await waitForLessBuffer();
+            }
+
+            if (websocket.readyState !== 1) {
+                return;
+            }
+
+            const entry = highPriorityQueue.shift() ?? lowPriorityQueue.shift();
```

Comment:
This way we actually flip the problem, because until we have high priority messages we don't process the low priority ones.

Thinking about using a round robin scheduling, perhaps by pulling 2 high priority messages and 1 low priority.
This way streams would still be sent but we leave more space for the more important updates.